src/mqtt.c

```diff
@@ -277,6 +286,26 @@ static int mqtt_connect (mqtt_client_conf_t *conf)
         return (-1);
     }
 
+#if LIBMOSQUITTO_MAJOR != 0
+    if (conf->cafile) {
+        status = mosquitto_tls_set(conf->mosq, conf->cafile, NULL,
+	                           conf->certfile, conf->keyfile, /* pw_callback */NULL);
+        if (status != MOSQ_ERR_SUCCESS) {
+            ERROR ("mqtt plugin: cannot mosquitto_tls_set: %s", mosquitto_strerror(status));
```

Comment:
I think we should destroy `conf->mosq` and return non-zero when any one of these functions fails.